build-support/src/main/kotlin/aws/sdk/kotlin/gradle/crt/CMakeTasks.kt

```diff
@@ -0,0 +1,202 @@
+
+            runCmake(project, knTarget, args)
+        }
+    }
+}
+
+internal fun Project.registerCmakeBuildTask(
+    knTarget: KotlinNativeTarget,
+    buildType: CMakeBuildType,
+): TaskProvider<Task> {
+    val cmakeBuildDir = project.cmakeBuildDir(knTarget)
+    val relativeBuildDir = cmakeBuildDir.relativeTo(project.rootDir).path
+
+    return project.tasks.register(knTarget.namedSuffix("cmakeBuild", capitalized = true)) {
+        group = "ffi"
+
+        inputs.property("buildType", buildType.toString())
+        inputs.file(project.cmakeLists)
+        inputs.files(
+            fileTree("$rootDir/crt").matching {
+                include(listOf("**/CMakeLists.txt", "**/*.c", "**/*.h"))
+            },
+        )
+
+        outputs.dir(project.cmakeBuildDir(knTarget))
+
+        doLast {
+            val args = mutableListOf(
+                "--build",
+                relativeBuildDir,
+                "--config",
+                buildType.toString(),
+            )
+
+            val osxSdk = knTarget.konanTarget.osxDeviceSdkName
+            if (osxSdk != null) {
+                // see https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html#switching-between-device-and-simulator
+                // assumes Xcode generator
+                args.add("--")
+                args.add("-sdk")
+                args.add(osxSdk)
+            }
+
+            runCmake(project, knTarget, args)
+        }
+    }
+}
+
+internal fun Project.registerCmakeInstallTask(
+    knTarget: KotlinNativeTarget,
+    buildType: CMakeBuildType,
+): TaskProvider<Task> {
+    val cmakeBuildDir = project.cmakeBuildDir(knTarget)
+    val relativeBuildDir = cmakeBuildDir.relativeTo(project.rootDir).path
+
+    return project.tasks.register(knTarget.namedSuffix("cmakeInstall", capitalized = true)) {
+        group = "ffi"
+
+        inputs.file(project.cmakeLists)
+        outputs.dir(project.cmakeInstallDir(knTarget))
+
+        doLast {
+            val args = mutableListOf(
+                "--install",
+                relativeBuildDir,
+                "--config",
+                buildType.toString(),
+            )
+            runCmake(project, knTarget, args)
+        }
+    }
+}
+
+internal fun runCmake(project: Project, target: KotlinNativeTarget, cmakeArgs: List<String>) {
+    project.exec {
+        workingDir(project.rootDir)
+        val exeArgs = cmakeArgs.toMutableList()
+        val exeName = when (target.konanTarget) {
+            KonanTarget.LINUX_X64, KonanTarget.LINUX_ARM64 -> {
+                // cross compiling via dockcross - set the docker exe to cmake
+                exeArgs.add(0, "cmake")
+                "./dockcross-" + target.konanTarget.name.replace("_", "-")
+            }
+            else -> "cmake"
+        }
+
+        project.logger.info("$exeName ${exeArgs.joinToString(separator = " ")}")
+        // FIXME - remove
+        println("$exeName ${exeArgs.joinToString(separator = " ")}")
```

Comment:
**Question**: Can this `FIXME` be addressed now? Or is the redundant `println` still needed for some reason?

Reply:
it can be removed now I was debugging something and didn't want the verbosity of `--info`.